lib/api/objectPut.js

```diff
@@ -211,74 +194,50 @@ function objectPut(authInfo, request, streamingV4Params, log, callback) {
-                                    return _storeIt(bucket, objectKey,
-                                        objMD, authInfo, canonicalID,
-                                        cipherBundle, request,
-                                        streamingV4Params, log,
-                                            (err, contentMD5) =>
-                                                callback(err, contentMD5,
-                                                    corsHeaders));
-                                });
-                        }
-                        return _storeIt(bucket, objectKey, objMD,
-                                        authInfo, canonicalID, null, request,
-                                        streamingV4Params, log,
-                                (err, contentMD5) =>
-                                    callback(err, contentMD5, corsHeaders));
-                    });
-        }
-        if (serverSideEncryption) {
-            return kms.createCipherBundle(
-                serverSideEncryption,
-                log, (err, cipherBundle) => {
-                    if (err) {
-                        return callback(errors.InternalError, null,
-                            corsHeaders);
-                    }
-                    return _storeIt(bucket, objectKey, objMD,
-                                    authInfo, canonicalID, cipherBundle,
-                                    request, streamingV4Params, log,
-                            (err, contentMD5) =>
-                                callback(err, contentMD5, corsHeaders));
-                });
-        }
-        return _storeIt(bucket, objectKey, objMD, authInfo, canonicalID,
-                        null, request, streamingV4Params, log,
-                        (err, contentMD5) =>
-                            callback(err, contentMD5, corsHeaders));
+        return async.waterfall([
+            callback => {
+                if (bucket.hasTransientFlag() || bucket.hasDeletedFlag()) {
+                    return cleanUpBucket(bucket, canonicalID, log, callback);
+                }
+                return callback();
+            },
+            callback => {
+                const serverSideEncryption = bucket.getServerSideEncryption();
+                if (serverSideEncryption) {
+                    return kms.createCipherBundle(
+                            serverSideEncryption, log, callback);
+                }
+                return callback(null, null);
+            },
+            (cipherBundle, callback) => createAndStoreObject(bucketName,
+                bucket, objectKey, objMD, authInfo, canonicalID, cipherBundle,
+                request, streamingV4Params, log, callback),
+        ], (err, res) => {
+            if (err) {
+                return callback(err, null, corsHeaders);
+            }
+            const newByteLength = request.parsedContentLength;
+            const oldByteLength = objMD ? objMD['content-length'] : null;
```

Comment:
If it's a new object, Utapi expects oldByteLength to be null. That means we need to check that `objMD['content-length']` is not `undefined`. See https://github.com/scality/S3/blob/rel/6.4/lib/api/objectPut.js#L79